lib/indexer/addrindexer.js

```diff
@@ -99,12 +102,26 @@ class AddrIndexer extends Indexer {
 
   async unindexBlock(entry, block, view) {
     const b = this.db.batch();
+
     for (let i = 0; i < block.txs.length; i++) {
       const tx = block.txs[i];
       const hash = tx.hash();
       for (const addr of tx.getHashes(view))
         b.del(layout.T.encode(addr, hash));
 
+      for (let i = 0; i < tx.outputs.length; i++) {
+        const output = tx.outputs[i];
+        const addr = output.getHash();
+
+        if (!addr)
+          continue;
+
+        b.del(layout.C.encode(addr, hash, i));
+      }
+    }
+
+    for (let i = 0; i < block.txs.length; i++) {
+      const tx = block.txs[i];
       if (!tx.isCoinbase()) {
         for (const {prevout} of tx.inputs) {
           const {hash, index} = prevout;
```

Comment:
We need to make sure inputs are not part of the current block.

Reply:
Thanks, fixed.